### ISSUE TITLE
Implement GetSymbolStoreFileName

### ISSUE THREAD

**Issue by tmat** [dotnet/symreader-portable#50]
Tracking TODO in source

**Comment by KirillOsenkov** [comment]
Do you have any tips on how this should be implemented? 
https://github.com/dotnet/symreader-portable/blob/7380d910e1b06b19c665ad74dc0902c2e5963f8e/src/Microsoft.DiaSymReader.PortablePdb/SymReader.cs#L685

Just return the full path to the .dll file?

**Comment by tmat** [comment]
Pretty much reverse-engineer native SymReader and do what it does.

**Comment by KirillOsenkov** [comment]
Do you have a pointer to the native source of GetSymbolStoreFileName? Where can I find it?

This exception would be nice to fix soon, I'm running into this quite a bit.

**Comment by KirillOsenkov** [comment]
CDiaWrapper.GetPdbName does this:
```
HRESULT
CDiaWrapper::GetPdbName(
    _Deref_out_z_ wchar_t** pdszPdbName
)
{
    HRESULT hr = S_OK;
    CComPtr< IDiaSymbol > pSymbol;
    CDiaString pdbname;

    IfFailGo( GetExeSymbol( &pSymbol ) );

    IfFailGo( pSymbol->get_symbolsFileName( &pdbname ));
    *pdszPdbName = pdbname.Detach();

Error:
    return hr;
}
```

**Comment by KirillOsenkov** [comment]
https://msdn.microsoft.com/en-us/library/075yb9bb.aspx

**Comment by KirillOsenkov** [comment]
I poked around a little bit but couldn't find an easy way to get at the full path to the .pdb (or .dll/.exe in case of embedded .pdb).

Feels like to implement this one would need to thread the filepath through somehow. Not sure what's the best way to do it.

**Comment by tmat** [comment]
Where is this API used? 

Some APIs on ISymUnmanagedBinder take the path. So they should probably pass it to SymReader ctor as an optional parameter, so that SymReader can return the path.

**Comment by KirillOsenkov** [comment]
One sample usage is in Mdbg:

```
System.NotImplementedException: The method or operation is not implemented.
   at int Microsoft.DiaSymReader.PortablePdb.SymReader.GetSymbolStoreFileName(int bufferLength, out int count, out char[] name)
   at string Microsoft.Samples.Debugging.CorSymbolStore.SymReader.GetSymbolStoreFileName()
   at string Microsoft.Samples.Debugging.MdbgEngine.ManagedModule.get_SymbolFilename()
```

When creating a ManagedModule they access and store the .pdb file path.

I was thinking about passing through an optional path from the creators of SymReader. It felt a bit dirty as otherwise SymReader is fully isolated from where the symbols came from (stream, file, etc). Maybe still worth it.

**Comment by KirillOsenkov** [comment]
Yup, just worked around it. If I know that the API will throw, why call it in the first place. Going to live with a patched Mdbg for a while.
